src/simpleNbox.cpp

```diff
@@ -595,6 +596,15 @@ unitval SimpleNbox::getData(const std::string& varName,
     } else if( varNameParsed == D_LUC_EMISSIONS ) {
         H_ASSERT( date != Core::undefinedIndex(), "Date required for luc emissions" );
         returnval = lucEmissions.get( date );
+    } else if( varNameParsed == D_CA_CONSTRAIN ) {
+        H_ASSERT( date != Core::undefinedIndex(), "Date required for atmospheric CO2 constraint" );
+        if (Ca_constrain.exists(date)) {
+            returnval = Ca_constrain.get( date );
+        } else {
+            H_LOG( logger, Logger::WARNING ) << "No CO2 constraint for requested date " << date <<
```

Comment:
If it were me, I'd log this at no higher than `NOTICE` (and I'd probably opt for `DEBUG`).  The reason why is that as things stand, the easiest way to figure out what dates have constraints is to just ask for all dates and check to see which ones do or don't come back with `NA`.  That's a legitimate use case that shouldn't generate a bunch of warnings (which, to me mean "probably wrong, but not necessarily or fatally so").  

Alternatively, we could implement a function in the API to check what dates have valid constraints, but I don't think that will necessarily be any easier for users (particularly R users, for whom it's pretty natural to pass big blocks of dates and drop `NA`s).  Plus, that will get kind of ugly when we add all of the other constraint options in #347 .

Reply:
Changed to `Logger::NOTICE` in 80eb6b7.

Re: API for missing constraints -- I agree that it's probably easiest just to check for `NA` on the R side.